core/charm/repository/charmhub.go

```diff
@@ -425,20 +425,26 @@ func (c *CharmHubRepository) getEssentialMetadataForBatch(reqs []corecharm.Metad
 
 	var metaRes = make([]corecharm.EssentialMetadata, len(reqs))
 	for resIdx, refreshResult := range refreshResults {
-		if len(refreshResult.Entity.MetadataYAML) == 0 {
+		if refreshResult.Entity.MetadataYAML == "" {
 			return nil, errors.Errorf("charmhub refresh response for %q does not include the contents of metadata.yaml", reqs[resIdx].CharmURL)
 		}
 		chMeta, err := charm.ReadMeta(strings.NewReader(refreshResult.Entity.MetadataYAML))
 		if err != nil {
 			return nil, errors.Annotatef(err, "parsing metadata.yaml for %q", reqs[resIdx].CharmURL)
 		}
 
-		if len(refreshResult.Entity.ConfigYAML) == 0 {
-			return nil, errors.Errorf("charmhub refresh response for %q does not include the contents of config.yaml", reqs[resIdx].CharmURL)
-		}
-		chConfig, err := charm.ReadConfig(strings.NewReader(refreshResult.Entity.ConfigYAML))
-		if err != nil {
-			return nil, errors.Annotatef(err, "parsing config.yaml for %q", reqs[resIdx].CharmURL)
+		// It's okay for a charm not to have a config.yaml. But the CharmHub
+		// API currently returns "{}\n" for charms that have no config.yaml,
+		// so handle that as no config as well.
+		var chConfig *charm.Config
+		configYAML := refreshResult.Entity.ConfigYAML
+		if configYAML == "" || strings.TrimSpace(configYAML) == "{}" {
```

Comment:
I wonder if we want a small helper since this check is done elsewhere too.
Maybe not worth it?

Reply:
Not sure it's worth it at this point, but happy to do so if there are more.